### PR TITLE
Jsui553 search alerts

### PR DESCRIPTION
**Pull request by mcarpentierCoveo**

### CODE REVIEW

File: src/ui/SearchAlerts/FollowItem.ts [hunk 1]
```diff
@@ -34,13 +42,13 @@ module Coveo {
...
-      this.container.append(this.text);
-      this.container.on('click', $.proxy(this.toggleFollow, this));
+      this.container = $$(this.element);
+      this.text = $$('span');
+      this.container.append(this.text.el);
+      this.container.on('click', ()=>{this.toggleFollow()});
 
-      $(this.root).on(SearchAlertEvents.searchAlertDeleted, $.proxy(this.handleSubscriptionDeleted, this));
-      $(this.root).on(SearchAlertEvents.searchAlertCreated, $.proxy(this.handleSubscriptionCreated, this));
+      $$(this.root).on(SearchAlertEvents.searchAlertDeleted, (e: Event, args: ISearchAlertEventArgs)=>{this.handleSubscriptionDeleted(e, args)});
```
Comment (olamothe) [inline src/ui/SearchAlerts/FollowItem.ts]: When inside the scope of a Component, you should always use `this.bind.onRootElement`

`this.bind` refers to the `ComponentEvents` class, which ensure that event handler are called only if the component is not disabled. Also, the useless `Event` parameters is not passed when using that method.

File: src/ui/SearchAlerts/SearchAlerts.ts [hunk 2]
```diff
@@ -1,298 +1,319 @@
...
-            subscription.id = updatedSearchAlert.id;
-            var eventArgs: SearchAlertEventArgs = { subscription: subscription };
-            $(this.root).trigger(SearchAlertEvents.searchAlertCreated, eventArgs);
-          })
-          .catch(() => {
-            this.handleSearchAlertsFail();
-          })
-      });
+  private addSearchAlert(subscription: ISubscription, container: Dom) {
+    var frequencies = [
```
Comment (olamothe) [inline src/ui/SearchAlerts/SearchAlerts.ts]: replace all `var` by the `let` keyword.

File: src/ui/SearchAlerts/SearchAlerts.ts [hunk 3]
```diff
@@ -1,298 +1,319 @@
...
+  private updateAndSyncSearchAlert(subscription: ISubscription) {
+    this.queryController.getEndpoint()
+      .updateSubscription(subscription)
+      .then((updated: ISubscription) => _.extend(subscription, updated))
+      .catch(() => {
+        this.handleSearchAlertsFail();
+      });
+  }
+
+  // FollowQuery
```
Comment (olamothe) [inline src/ui/SearchAlerts/SearchAlerts.ts]: Remove this comment and add the jsdoc for those component. They will need to be documented like all other component. 

This means all options and public methods.

You can also reference this https://coveord.atlassian.net/browse/JSUI-770 in your commit message.

**Comment by olamothe** [comment]
You can add assignees to pull requests (right sidebar in github UI).

**Comment by mcarpentierCoveo** [comment]
@olamothe can i :shipit:

**Comment by olamothe** [comment]
:lgtm:

Sorry about the delay !

<details><summary><i>Previously, mcarpentierCoveo (Mathieu Carpentier) wrote…</i></summary>
> @olamothe can i :shipit: 
</details>


---

Reviewed 3 of 12 files at r1, 3 of 9 files at r2, 6 of 6 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/coveo/search-ui/11#-:-KJkKHHkY7LSSjhEAa96:-1594846244)*

**Pull request merged**